src/wanna/components/kubeflow/get_or_create_endpoint.py

```diff
@@ -0,0 +1,66 @@
+from typing import Dict
+
+from kfp.v2 import dsl
+
+
+@dsl.component(
+    base_image="python:3.9",
+    packages_to_install=[
+        "google-cloud-pipeline-components==1.0.14",
+        "google-cloud-aiplatform==1.15.1",
+    ],
+)
+def get_or_create_endpoint(
+    project: str,
+    location: str,
+    display_name: str,
+    endpoint: dsl.Output[dsl.Artifact],
```

Comment:
dsl.Artifact should be `google.VertexEndpoint`

Reply:
Based on google_cloud_pipeline_components.v1.endpoint.EndpointCreateOp return type documented in https://google-cloud-pipeline-components.readthedocs.io/en/google-cloud-pipeline-components-1.0.14/google_cloud_pipeline_components.v1.endpoint.html#module-google_cloud_pipeline_components.v1.endpoint

Reply:
This is unfortunately not working currently, kubeflow fails because of the "google." namespace. 

They currently recommend:

https://cloud.google.com/vertex-ai/docs/pipelines/use-components#use_python_function-based_components
# Note currently KFP SDK doesn't support outputting artifacts in `google` namespace.
# Use the base type dsl.Artifact instead.

It has the same functionality, I tried it with sushi and sklearn example pipelines and it works. The only difference is in the UI, that you see an Artifact with name endpoint and not google.VertexEndpoint.

Reply:
Or we could make it a yaml based component with a container, but I found this one easiers. Then with yaml we could have google.VertexEndpoint